toolchain/static-link.lisp

```diff
@@ -33,6 +33,18 @@
 
 ;; Problem? Its output may conflict with the program-op output :-/
 
+(defmethod perform :before ((op static-image-op) (s system))
+  ;; Close non-system foreign libraries, as grovel ones are embedded
+  ;; in the static-image. System foreign libraries still need to be
+  ;; explicitly loaded on startup, so we're keeping them loaded.
+  (register-image-dump-hook
+   (lambda ()
+     (loop
+        for library in (list-foreign-libraries)
+        when (eql (foreign-library-type library) :grovel-wrapper)
+        do (close-foreign-library library))))
+  (setf *image-entry-point* (ensure-function (asdf/system:component-entry-point s))))
```

Comment:
What does this line do?

Reply:
Given that I'm overriding `defmethod perform :before` from `program-op` (see `bundle.lisp`), I need to re-do it. It's most likely lack of Lisp knowledge from my part, as I tried calling `(call-next-method)`, unsuccessfully.

Reply:
All applicable :before methods will run in most-specific-first order, so there's no overriding going on.

Reply:
Oh, I'm redefining the PERFORM method on STATIC-IMAGE-OP instead of STATIC-PROGRAM-OP...

Reply:
Fixed now.